lib/util/progress.coffee

```diff
@@ -179,8 +179,8 @@ module.exports =
       if not @stack.length
         @tooltip.hide_()
       # backwards iteration
-      for p in @stack.reverse()
-        table.appendChild @tableRowView p
+      for i in [0...@stack.length].reverse()
+        table.appendChild @tableRowView @stack[i]
```

Comment:
`p in array` doesn't guarantee any special order: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in
Tripped me up as well when I first implemented this...